### ISSUE TITLE
Question: Order of stacked bars/areas vs order of legend

### ISSUE THREAD

**Issue by Eric24** [recharts/recharts#571]
Is it possible to control the order that a datakey appears in a stacked bar/area independently of the order it appears in the legend (without resorting to a custom legend)? I can "flip" both by controlling the order of the Bar or Area components in the declaration, but the order of the stack is always the exact opposite of the order the corresponding item appears in the legend. See the example below (same for a vertical chart with a vertical legend).

![image](https://cloud.githubusercontent.com/assets/685316/23881235/2e395b70-0827-11e7-99c7-3b0355b2407b.png)

**Comment by anthony-viavoo** [comment]
As workaround, I create <Legend> payload manually using .reverse() from data used to create my Bar. Fortunately, I use immutable.js to manage this data.

```
const payload = data.reverse().map((legendItem) => ({
    value: legendItem.value,
    id: legendItem.id,
    type: 'square',
});
return <Legend payload={payload} />;
```

**Comment by ador** [comment]
This bothers me as well...

**Comment by ador** [comment]
Same problem as here: https://github.com/recharts/recharts/issues/209

**Comment by swarthy** [comment]
:+1:

**Comment by smishr4** [comment]
Has anyone figured out a way to solve this?

**Comment by Eric24** [comment]
This has been fixed (works in 1.5, maybe earlier).
